utils_gorm.go

```diff
@@ -0,0 +1,181 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/iver-wharf/wharf-core/pkg/ginutil"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+func findDBPaginatedSliceAndTotalCount(dbQuery *gorm.DB, limit, offset int, slice interface{}, totalCount *int64) error {
```

Comment:
nit: rename `slice` -> `slicePtr` as that is what is being passed to the function when used throughout the code. Good to be explicit when dealing with `interface{}`.
Not sure if gorm's `Find` deals with it so that either is fine, but feels scary to rely on that.
```suggestion
func findDBPaginatedSliceAndTotalCount(dbQuery *gorm.DB, limit, offset int, slicePtr interface{}, totalCount *int64) error {
```

Reply:
Good suggestion! I don't know either, but I'm assuming when it needs to enlarge the underlying array the lack of pointer makes it "pointless"